erpnext/stock/doctype/quality_inspection/quality_inspection.json

```diff
@@ -226,12 +244,99 @@
    "label": "Status",
    "options": "\nUntested\nAccepted\nRejected\nPartially Rejected\nSkipped",
    "reqd": 1
+  },
+  {
+   "fieldname": "section_break_strain_test",
+   "fieldtype": "Section Break",
+   "label": "Strain Test"
+  },
+  {
+   "fieldname": "strain_notes",
+   "fieldtype": "Text",
+   "label": "Strain Notes"
+  },
+  {
+   "fieldname": "testing_lab",
+   "fieldtype": "Link",
+   "label": "Testing Lab",
+   "options": "Supplier"
+  },
+  {
+   "fieldname": "column_break_strain_test",
+   "fieldtype": "Column Break"
+  },
+  {
+   "fieldname": "testing_result_link",
+   "fieldtype": "Data",
+   "label": "Testing Result Link"
+  },
+  {
+   "fieldname": "test_date",
+   "fieldtype": "Date",
+   "label": "Test Date"
+  },
+  {
+   "fieldname": "section_break_manufacture_information",
+   "fieldtype": "Section Break",
+   "label": "Manufacture Information"
+  },
+  {
+   "fieldname": "manufacturer_name",
+   "fieldtype": "Link",
+   "label": "Manufacturer Name",
+   "options": "Supplier"
+  },
+  {
+   "fieldname": "column_break_manufacture_information",
+   "fieldtype": "Column Break"
+  },
+  {
+   "fieldname": "manufacturer_website",
+   "fieldtype": "Data",
+   "label": "Manufacturer Website"
```

Comment:
Are these being pulled from anywhere, or should they be editable?

Reply:
"Manufacturer Name" is pulled from Supplier DocType and The "Manufacturer Website" is simple editable data field according to DoD

Reply:
Make both these fields read-only, and:

- Pull "Manufacturer Name" as the Supplier field from the linked Purchase document
- Pull "Manufacturer Website" as the website field from the linked Supplier